research/algorithm1.md

```diff
@@ -0,0 +1,35 @@
+Author: Tran Bao Phuc Long - 2352703 & Nguyen Thanh Hieu - 2352331
+The heuristic approach is widely used in complex optimization problems like the cutting stock problem due to its balance of efficiency and effectiveness. Here’s an in-depth look at the heuristic approach, its principles, advantages, and disadvantages.
+
+### **What is a Heuristic Approach?**
+
+A heuristic approach in optimization is a rule-based method that aims to find a "good enough" solution within a reasonable time frame. Unlike exact algorithms that aims to search for the best solution, heuristics provide practical solutions by employing strategies such as prioritization, approximation, and incremental refinement. For the two-dimensional cutting stock problem, a heuristic might involve:
+
+1. **Prioritization**: Sorting products by size, quantity, or other metrics to maximize efficient usage of stock material.
+2. **Placement Optimization**: Placing items in a way that maximizes the use of available stock, such as using a greedy or waste-minimization criterion.
+3. **Iterative Refinement**: Continuously adjusting placements to improve the layout and reduce leftover gaps.
+
+### **Key Components of Heuristic Approaches in Cutting Stock Problems**
+
+1. **Product Sorting**: Prioritizes products based on size or shape. Larger products are often placed first to reduce large, irregular gaps.
+2. **Greedy Placement**: Uses a local optimization strategy by placing items in the "best" current spot, defined by criteria like minimal waste or proximity to edges.
+3. **Waste Minimization**: Evaluates possible placements and chooses the one that leaves the least amount of unused stock.
+4. **Flexibility in Decision-Making**: Heuristic methods often make trade-offs between speed and optimality, focusing on practical solutions that are near-optimal but computed quickly.
+
+### **Advantages of the Heuristic Approach**
```

Comment:
Please prolong this section for document purpose

Reply:
Done, and also refined the disadvantages section.